tests/python/contrib/test_coreml_codegen.py

```diff
@@ -0,0 +1,64 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import pytest
+import numpy as np
+
+import tvm
+from tvm import relay
+from tvm.relay import transform
+from tvm.contrib.target import coreml as _coreml
+
+def check_result(mod, map_inputs, out_shape, result, tol=1e-3, target="llvm",
+                 ctx=tvm.cpu(), params=None):
+    def check_graph_runtime_result():
+        with relay.build_config(opt_level=3):
+            json, lib, param = relay.build(mod, target=target, params=params)
+        rt_mod = tvm.contrib.graph_runtime.create(json, lib, ctx)
+
+        for name, data in map_inputs.items():
+            rt_mod.set_input(name, data)
+        rt_mod.set_input(**param)
+        rt_mod.run()
+        out = tvm.nd.empty(out_shape, ctx=ctx)
+        out = rt_mod.get_output(0, out)
+
+        tvm.testing.assert_allclose(out.asnumpy(), result, rtol=tol, atol=tol)
+
+    check_graph_runtime_result()
+
+@pytest.mark.skip('skip because coremltools is not available in CI')
```

Comment:
Does this mean there will be no testing in CI for this change? I think we should find a way to test as much as possible or enable CI to perform some proper tests, otherwise we're not going to have confidence we don't break this integration. For instance, is there a way we could just test the compilation pathway?

Reply:
Updated so that we check the test when coremltools is available.  We have coremltools in the ci-gpu, so this test will be checked in our CI.